tests/test_search.py

```diff
@@ -333,7 +333,7 @@ def test_search(query, expected):
                 },
                 {
                     "A": "cat1",
-                    "B": ["a", "c"],
+                    "B": ["c", "a"],
```

Comment:
When searching over list columns in polars, the ordering of the initial list column is preserved. 

Whether we would prefer to preserve the initial list order or sort in order of the query as the previous implementation did is an open question - I can see arguments for both.

I've altered the test as I don't think it's important & it reduces the amount of data processing necessary.